opacus/grad_sample/grad_sample_module.py

```diff
@@ -425,6 +425,35 @@ def rearrange_grad_samples(
 
         return activations, backprops
 
+    def state_dict(self, *args, **kwargs) -> Dict:
+        """
+        Return the state dict of the wrapped module
+        """
+        ret_state_dict = {
+            f"_module.{key}": value
+            for key, value in self._module.state_dict(*args, **kwargs).items()
+        }
```

Comment:
Are you sure about this?
In my experience "_module" prefix is always an annoyance and never helps. When I was mentioning the discrepancy in my previous comments, I meant that we should approach it from a different end and remove the prefix everywhere. WDYT?

In particular, getting rid of the prefix makes it much easier to load weights from publicly pre-trained models (currently you have to do it strictly before you wrap a model with GSM)

Reply:
> in my previous comments, I meant that we should approach it from a different end and remove the prefix everywhere. WDYT?
Ah! That works too. I don't have a strong preference so long as it is consistent. I'll change that.

> In my experience "_module" prefix is always an annoyance and never helps.

Out of curiosity, what trouble ddi the prefix cause in the past>

Reply:
Scenario 1: You have a "--disable-dp" flag in your pipeline, and you want to train and compare dp and non-dp model. Imagine also, that your infra is unstable, so you checkpoint the model often and run evaluation in a separate job. You then need to handle two different formats of state_dicts
Scenario 2: In general, when you train DP model and then want to transform it to inference model (non-DP obviously), you'd rather have same state_dicts
Scenario 3: When doing public pretraining, today you have to be careful and load_state_dict before you wrap the model with GradSampleModule. With same state_dicts you don't have to worry about it

Reply:
JFYI, to clarify things, `state_dict` and `load_state_dict` are already available in GradSampleModule today, as it's inherited from nn.Module, and it behaves exactly like this, so there's no need to override the method even. 

To be fair, one argument for leaving this alone for now is:
a) it doesn't directly help the task at hand, checkpointing can work with either options
b) it will break backward compatibility for some use cases, and even though I do believe it's a change for good, we might want to approach this more carefully.

Before moving forward, let's do one of the following:
1) Decide to deal with parameter names later and remove `state_dict` and `load_state_dict` overrides (because default inherited behaviour is identical)
2) Decide to rename parameters in state_dict now and don't add "_module." prefix

I'm fine with either, and for me that's the only outstanding item before accepting